src/figury/AnimatorApp.java

```diff
@@ -76,7 +77,24 @@ public void actionPerformed(ActionEvent e) {
 		});
 		btnAnimate.setBounds(100, 239, 80, 23);
 		contentPane.add(btnAnimate);
-		
+
+
+		contentPane.addComponentListener(new ComponentAdapter() {
+			@Override
+			public void componentResized(ComponentEvent e) {
+						super.componentResized(e);
+						Component eComponent = e.getComponent();
+						int w = eComponent.getWidth();
+						int h = eComponent.getHeight();
+						System.out.println("Zmieniono rozmiar");
+						kanwa.setSize(w - 30, h - 80);
```

Comment:
to nie takie proste - kanwa zrobiła już wcześniej bufor o jakichś rozmiarach, figury mają też ten bufor...